checkout/src/main/java/com/checkout/tokenization/usecase/RiskSdkUseCase.kt

```diff
@@ -0,0 +1,55 @@
+package com.checkout.tokenization.usecase
+
+import android.content.Context
+import com.checkout.base.model.Environment
+import com.checkout.base.usecase.UseCase
+import com.checkout.risk.Risk
+import com.checkout.risk.RiskConfig
+import com.checkout.risk.RiskEnvironment
+import com.checkout.tokenization.model.TokenDetails
+import com.checkout.tokenization.model.TokenResult
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+internal class RiskSdkUseCase(
+    private val environment: Environment,
+    private val context: Context,
+    private val publicKey: String,
+    private val riskInstanceProvider: RiskInstanceProvider = RiskInstanceProvider(),
+) : UseCase<TokenResult<TokenDetails>, Unit> {
+    override fun execute(data: TokenResult<TokenDetails>) {
+        CoroutineScope(Dispatchers.IO).launch {
+            val riskInstance = riskInstanceProvider.provide(context, publicKey, environment)
+            when (data) {
+                is TokenResult.Success -> {
+                    riskInstance?.publishData(cardToken = data.result.token)
+                }
+                is TokenResult.Failure -> {}
+            }
+        }
+    }
+}
+
+internal class RiskInstanceProvider() {
```

Comment:
The purpose of having a provider is to create and reuse the instance all around the code, separating the login on how to set it up and being able to write tests around it.
By having ` RiskInstanceProvider()` as an internal class in the RiskSDKUseCase all of the above cannot be fulfilled.
Please have a look at `checkout/src/main/java/com/checkout/logging` and how we have the logger provider and interfaces over there. It also has some good test cases that can used as inspiration for this job